cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/UsageHandler.java

```diff
@@ -55,7 +59,8 @@ public void getAppDatasetUsage(HttpRequest request, HttpResponder responder,
                                  @PathParam("app-id") String appId) {
     final ApplicationId id = new ApplicationId(namespaceId, appId);
     Set<? extends EntityId> ids = registry.getDatasets(id);
-    responder.sendJson(HttpResponseStatus.OK, ids);
+    responder.sendJson(HttpResponseStatus.OK,
+                       Collections2.transform(ids, BackwardCompatibility.MAKE_COMPATIBLE_WITH_35));
```

Comment:
isn't transform lazy? It's not safe to use this when you want to serialize things right?